db/document.go

```diff
@@ -997,7 +997,9 @@ func (doc *Document) UnmarshalWithXattr(data []byte, xdata []byte, unmarshalLeve
 func (doc *Document) MarshalWithXattr() (data []byte, xdata []byte, err error) {
 	// Grab the rawBody if it's already marshalled, otherwise unmarshal the body
 	if doc._rawBody != nil {
-		data = doc._rawBody
+		if !doc.Deleted && !doc.IsDeleted() {
```

Comment:
Similar to the above question, it makes me nervous that we're relying on both doc.Deleted and doc.IsDeleted being set here.